src/frontend/screens/Audio/CreateRecording/index.tsx

```diff
@@ -1,40 +1,56 @@
 import React, {useEffect} from 'react';
-import {Text, View, StyleSheet} from 'react-native';
-import {WHITE} from '../../../lib/styles';
-
+import {BackHandler} from 'react-native';
 import {useNavigationFromRoot} from '../../../hooks/useNavigationWithTypes';
-import {CustomHeaderLeft} from '../../../sharedComponents/CustomHeaderLeft';
+import {RecordingActive} from './RecordingActive';
+import {RecordingDone} from './RecordingDone';
+import {RecordingIdle} from './RecordingIdle';
+import {useAudioRecording} from './useAudioRecording';
 
 export function CreateRecording() {
   const navigation = useNavigationFromRoot();
+  const {startRecording, stopRecording, reset, status, uri} =
+    useAudioRecording();
+
+  const currentState = status?.isRecording ? 'active' : uri ? 'done' : 'idle';
+
   useEffect(() => {
-    navigation.setOptions({
-      headerLeft: props => (
-        <CustomHeaderLeft
-          tintColor={props.tintColor}
-          headerBackButtonProps={props}
-        />
-      ),
-    });
-  }, [navigation]);
-  return (
-    <View style={styles.contentContainer}>
-      <View style={styles.container}>
-        <Text style={styles.message}>Create Recording</Text>
-      </View>
-    </View>
-  );
-}
+    if (currentState === 'active' || currentState === 'done') {
+      const onBackPress = () => {
+        return true;
+      };
+      const backHandler = BackHandler.addEventListener(
+        'hardwareBackPress',
+        onBackPress,
+      );
+      return () => {
+        backHandler.remove();
+      };
+    }
+  }, [currentState]);
 
-const styles = StyleSheet.create({
-  contentContainer: {flex: 1},
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-  },
-  message: {
-    color: WHITE,
-    fontSize: 20,
-    textAlign: 'center',
-  },
-});
+  if (currentState === 'idle') {
+    return <RecordingIdle onPressRecord={startRecording} />;
+  }
+
+  if (currentState === 'active') {
+    return (
+      <RecordingActive
+        duration={status?.durationMillis || 0}
+        onPressStop={stopRecording}
+      />
+    );
+  }
+
+  if (currentState === 'done') {
+    return (
+      <RecordingDone
+        uri={uri || ''}
+        duration={status?.durationMillis || 0}
+        onDelete={() => {
+          reset();
+        }}
+        onRecordAnother={reset}
```

Comment:
onDelete and onReset are now the same thing. It would just be clearer is reset was just a prop of RecordingDone and just pass reset once